types/recompose/index.d.ts

```diff
@@ -268,20 +269,22 @@ declare module 'recompose' {
 
     // Static property helpers: https://github.com/acdlite/recompose/blob/master/docs/API.md#static-property-helpers
 
+    type ComponentModifier = <T extends Component>(component: T) => T;
```

Comment:
Why introduce a new type, rather than improving `ComponentEnhancer`?  (It seems like the goal was to reflect the fact that the returned function maps `StatelessComponent`s to `StatelessComponent`s.)

Reply:
`ComponentEnhancer` is used for the HOC functions provided by `recompose`.  The HOC functions return functions that wrap a component within an outer `ComponentClass` component ([example](https://github.com/acdlite/recompose/blob/5b8498a48e8f7ad9dd6b2706be2fc3311d86543c/src/packages/recompose/renameProp.js#L12-L15)).

The static property helpers provided by `recompose` behave differently.  They return functions that simply set a static property and return the component they were called with.

Reply:
I don't know enough about react to understand the distinction you're making, but I'm satisfied that you've thought of it.  My context-less follow-up question would be whether some of the other `ComponentEnhancer` consumers should be updated to use `ComponentModifier`?  If might also be helpful to document the two types in such a way that future consumers can decide which is appropriate for their use case.